src/fmeval/eval_algorithms/general_semantic_robustness.py

```diff
@@ -330,18 +330,17 @@ def evaluate(
             dataset_prompt_template = (
                 get_default_prompt_template(dataset_config.dataset_name) if not prompt_template else prompt_template
             )
-            model_invocation_pipeline = create_model_invocation_pipeline(model, dataset_prompt_template)
-            dataset = model_invocation_pipeline.execute(dataset)
-            is_deterministic = verify_model_determinism(model, dataset, DatasetColumns.PROMPT.value.name)
-            pipeline = self._build_pipeline(model, dataset_prompt_template, is_deterministic=is_deterministic)
-            eval_output = compute_and_aggregate_metrics(
-                pipeline=pipeline,
+            is_deterministic = verify_model_determinism(model, dataset, dataset_prompt_template)
```

Comment:
Note: We need to do a model check before this for semantic robustness algorithms.

Reply:
Since the `model` argument is not `Optional`, I will omit the explicit check. However, I have updated the docstring to indicate the reasoning for why the model is required. The same docstring update has been applied to the other SR algos.